internals-js/src/specs/sourceSpec.ts

```diff
@@ -456,17 +476,21 @@ export class SourceSpecDefinition extends FeatureDefinition {
       // @sourceField is allowed only on root Query and Mutation fields or
       // fields of entity object types.
       const fieldParent = application.parent;
-      if (fieldParent.sourceAST?.kind !== "FieldDefinition") {
+      if (fieldParent.sourceAST?.kind !== Kind.FIELD_DEFINITION) {
         errors.push(ERRORS.SOURCE_FIELD_NOT_ON_ROOT_OR_ENTITY_FIELD.err(
           `${sourceField} must be applied to field`,
-          { nodes: application.sourceAST },
+          { nodes: [application.sourceAST!, fieldParent.sourceAST!] },
         ));
       } else {
         const typeGrandparent = fieldParent.parent as SchemaElement<any, any>;
-        if (typeGrandparent.sourceAST?.kind !== "ObjectTypeDefinition") {
+        if (typeGrandparent.sourceAST?.kind !== Kind.OBJECT_TYPE_DEFINITION) {
           errors.push(ERRORS.SOURCE_FIELD_NOT_ON_ROOT_OR_ENTITY_FIELD.err(
             `${sourceField} must be applied to field of object type`,
-            { nodes: application.sourceAST },
+            { nodes: [
```

Comment:
What does the error look like when you include the parent and grandparent? I'm concerned that it might be confusing. I think typically we just want to point them to the line that needs to change (i.e. just `application.sourceAST`)

Reply:
The `nodes` property seems to get attached to the `GraphQLError` object, as given, so you'd have to examine that property to notice it. That said, I can change this back to `nodes: application.sourceAST` for consistency with the other errors.

Reply:
https://github.com/apollographql/federation/pull/2914/commits/16a346b3188ffb3ca3412242dfc944dc6cc39577

Reply:
That was my only comment, so if you make those changes, you can merge it.